packages/pharos/src/components/table/pharos-table.ts

```diff
@@ -0,0 +1,202 @@
+import { PharosElement } from '../base/pharos-element';
+import ScopedRegistryMixin from '../../utils/mixins/scoped-registry';
+import { html, nothing } from 'lit';
+import { state } from 'lit/decorators.js';
+import type { TemplateResult, CSSResultArray } from 'lit';
+import { property } from 'lit/decorators.js';
+import { tableStyles } from './pharos-table.css';
+import { PharosSelect } from '../select/pharos-select';
+import { PharosPagination } from '../pagination/pharos-pagination';
+
+/**
+ * Pharos table component.
+ *
+ * @tag pharos-table
+ *
+ * @fires pharos-table-prev-page - Fires when the table goes to previous page
+ * @fires pharos-table-next-page - Fires when the table goes to next page
+ *
+ */
+export class PharosTable extends ScopedRegistryMixin(PharosElement) {
+  static elementDefinitions = {
+    'pharos-select': PharosSelect,
+    'pharos-pagination': PharosPagination,
+  };
+
+  /**
+   * Coloumn definition.
+   * @attr columns
+   * sample column:
+   * [
+   *  {
+   *    name: ID
+   *    field: id
+   *  },
+   *  {
+   *    name: Message
+   *    field: message
+   *  }
+   * ]
+   */
+  @property({ type: Array, reflect: true, attribute: 'columns' })
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  public columns: any[] = [];
```

Comment:
Updated!

Reply:
I've been thinking about your suggestion Dane. I wonder if it is possible in some cases the content in a column can be in multiple types. Do we require the consumer to provide all possible types? Will that be too heavy load for the consumer?

Reply:
I would generally hope that if we're displaying tabular data, that within a column the data type is consistent (or can be coerced to be consistent by way of `string`, minimally). There's a tradeoff here between safety and tedium as you rightly call out...I don't really have strong feelings on this, and it can definitely be iterated on, so I don't want it to hold us up here too much!